src/Time/Actions/ChangeSlabSize.hpp

```diff
@@ -156,13 +156,22 @@ struct ChangeSlabSize {
             ? current_slab.with_duration_from_start(new_slab_size)
             : current_slab.with_duration_to_end(new_slab_size);
 
-    const auto new_step = current_step.with_slab(new_slab);
-
     // We are at a slab boundary, so the substep is 0.
     const auto new_time_step_id =
         TimeStepId(time_step_id.time_runs_forward(), time_step_id.slab_number(),
                    time_step_id.step_time().with_slab(new_slab));
 
+    TimeDelta new_step;
+    if constexpr (Metavariables::local_time_stepping) {
+      // choose the acceptable step in the new slab that as closely as possible
+      // matches the previous step value
+      const auto& step_controller = db::get<Tags::StepController>(box);
+      new_step = step_controller.choose_step(new_time_step_id.step_time(),
+                                             current_step.value());
```

Comment:
This is susceptible to roundoff error causing step-size fluctuations.  I suggest preventing the slab size from increasing rapidly instead of trying to deal with the fallout when it does.  (There is already code to do this: add the "Increase" StepChooser to your input file.)